src/sms/adapters/inbound/fastapi_/routes.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2021 - 2024 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""FastAPI routes."""
+
+from collections.abc import Mapping
+from typing import Annotated, Any
+
+from fastapi import APIRouter, Request, status
+from fastapi.exceptions import HTTPException
+
+from sms.adapters.inbound.fastapi_ import dummies
+from sms.adapters.inbound.fastapi_.http_authorization import (
+    TokenAuthContext,
+    require_token,
+)
+from sms.models import Criteria, DocumentType, UpsertionDetails
+from sms.ports.inbound.docs_handler import DocsHandlerPort
+
+router = APIRouter()
+
+
+@router.get(
+    "/health",
+    summary="health",
+    tags=["StateManagementService", "sms-"],
+    status_code=200,
+)
+async def health():
+    """Used to test if this service is alive"""
+    return {"status": "OK"}
+
+
+@router.get(
+    "/docs/permissions",
+    tags=["StateManagementService", "sms-mongodb"],
+    summary="Returns the configured db permissions list.",
+    status_code=200,
+)
+async def get_configured_permissions(
+    config: dummies.ConfigDummy,
+) -> list[str]:
+    """Returns the configured db permissions list."""
+    return config.db_permissions or []
+
+
+@router.get(
+    "/docs/{db_name}/{collection}",
```

Comment:
@Cito I think that depends on what's more ergonomic from the perspective of the API consumers. Pymongo doesn't support namespace style access to collections, so your proposed change would exist at point of the API, but the parameters would be have to be parsed out anyway.

Reply:
Parsing it would be very simple: `db_name, collection = namespace.split('.', 1)`. The advantage is that this is already a well-established notation for specifying the fully qualified collection , it would look more like a normal REST endpoint with the name of the resource and a single identifier, and be more similar to the endpoints for other resources which also take only one path identifier.

Reply:
Btw, I think we should change `docs` to `documents`, because `docs` is already used for the interactive OpenAPI, and the other endpoints are also not abbreviated.

Reply:
`/documents/namespace` is the format as of [9728304](https://github.com/ghga-de/state-management-service/pull/1/commits/97283041a6e26d68b5287622c3650f2c432e9292)